phalcon/Storage/Adapter/AbstractAdapter.zep

```diff
@@ -163,12 +163,6 @@ abstract class AbstractAdapter implements AdapterInterface, EventsAwareInterface
 
         this->fire(this->eventType . ":beforeGet", key);
 
-        if (true !== this->has(key)) {
-            this->fire(this->eventType . ":afterGet", key);
-
-            return defaultValue;
-        }
-
         let content  = this->doGet(key);
```

Comment:
Add a check here after the `this->doGet(key)` for `false`. Then return the `defaultValue` if it is. Redis for isntance returns `false` if the operation is not successful. Then all we have to do is adjust each adapter's `doGet` to return `false` on failure if they do not already do so.